app/javascript/mastodon/actions/notifications.js

```diff
@@ -433,11 +434,12 @@ export const fetchNotificationRequestFail = (id, error) => ({
   error,
 });
 
-export const acceptNotificationRequest = id => (dispatch) => {
+export const acceptNotificationRequest = (id, count) => (dispatch) => {
```

Comment:
It might be cleaner to compute the count inside the action, based on the `id`. The Redux best-practices are that you should keep your action parameters as simple as possible.

This could be done by using `(dispatch, getState)`, then having something like `const count = selectNotificationCountForRequest(getState(), id)`.

This is more code, but is more in line with Redux best practices, so I am fine with both approaches.

Reply:
You mentioned `getState` but I had a hard time figuring out what that gives me / how I can query it. The redux docs are not very helpful in that regard :confused:

Reply:
`getState()` gives you the root state. Here, accessing the notification would be something like `getState().getIn(['notificationRequests', 'items', id, 'notifications_count'], 0)`

Reply:
This does not work, as `id` is not the list index here.

Renaud answered my questions privately. His approach did not work either.

*But* I was able to combine both suggestions to get to a working solution :slightly_smiling_face: 

Thank you both for your help, please have another look.

Reply:
Looks good to me!